src/librustc/mir/mod.rs

```diff
@@ -305,11 +305,7 @@ impl<'tcx> Body<'tcx> {
     pub fn vars_iter<'a>(&'a self) -> impl Iterator<Item = Local> + 'a {
         (self.arg_count + 1..self.local_decls.len()).filter_map(move |index| {
             let local = Local::new(index);
-            if self.local_decls[local].is_user_variable.is_some() {
-                Some(local)
-            } else {
-                None
-            }
+            self.local_decls[local].is_user_variable.is_some().to_option(local)
```

Comment:
This can just be `self.local_decls[local].is_user_variable.map(|| local)`, right?

Reply:
It has to be `.as_ref().map(|_| local)` instead, but that's still more concise :+1: